src/mod/applications/mod_callcenter/mod_callcenter.c

```diff
@@ -3240,16 +3245,27 @@ SWITCH_STANDARD_APP(callcenter_function)
 		agent_found = switch_true(p);
 	}
 
-	/* Stop member thread */
+	/* Stop member thread and announcement */
 	if (h) {
 		h->running = 0;
 	}
 
+	if (queue_announce) {
+		switch_ivr_stop_displace_session(member_session, queue_announce);
+	}
+
+	/* in the case queue was reloaded */
+	if ((queue = get_queue(queue_name)) && queue->announce && switch_channel_get_private(member_channel, queue->announce)) {
+		switch_ivr_stop_displace_session(member_session, queue->announce);
+	}
+
+	queue_rwunlock(queue);
```

Comment:
rwunlock will crash if this part returns NULL
```
(queue = get_queue(queue_name))
```

Shouldn't we use the unlock inside the if block?

Reply:
You're totally right. I fixed it.